File: crates/chidori-js/benchmarks/workloads/arith_loop.js
// Tight numeric loop — interpreter dispatch + integer/float arithmetic.
// Mirrors the `arith_loop` criterion micro-benchmark, scaled up so execution
// dominates process startup when run as a standalone script.
const N = 1_000_000;
let s = 0;
for (let i = 0; i < N; i++) {
  s += i * 2 - (i % 3);
}
console.log("RESULT=" + s);

File: crates/chidori-js/benchmarks/workloads/arith_loop.py
# Python twin of arith_loop.js — tight numeric loop (interpreter dispatch +
# arithmetic). All values stay far below 2^53, so exact Python ints match the
# exact JS doubles digit for digit.
N = 1_000_000
s = 0
for i in range(N):
    s += i * 2 - (i % 3)
print("RESULT=" + str(s))
